compiler/rustc_mir_build/src/thir/constant.rs

```diff
@@ -46,7 +46,10 @@ crate fn lit_to_const<'tcx>(
         (ast::LitKind::Int(n, _), ty::Uint(_)) | (ast::LitKind::Int(n, _), ty::Int(_)) => {
             trunc(if neg { (*n as i128).overflowing_neg().0 as u128 } else { *n })?
         }
-        (ast::LitKind::Float(n, _), ty::Float(fty)) => parse_float(*n, *fty, neg),
+        (ast::LitKind::Float(n, _), ty::Float(fty)) => match parse_float(*n, *fty, neg) {
+            Some(f) => f,
+            None => return Err(LitToConstError::Reported),
+        },
```

Comment:
Its probably more conventional to write this like such:  

```suggestion
        (ast::LitKind::Float(n, _), ty::Float(fty)) => parse_float(*n, *fty, neg).ok_or(LitToConstError::Reported)?,
```